packages/ui/src/ui/store/actions/navigation/modals/cell-preview.ts

```diff
@@ -0,0 +1,94 @@
+import {ThunkAction} from 'redux-thunk';
+import {RootState} from '../../../reducers';
+import {ytApiV4} from '../../../../rum/rum-wrap-api';
+import {getPath} from '../../../selectors/navigation';
+import {batch} from 'react-redux';
+import {getDefaultRequestOutputFormat} from '../../../../utils/navigation/content/table/table';
+import type {CellPreviewAction} from '../../../reducers/navigation/modals/cell-preview';
+import {CELL_PREVIEW} from '../../../../constants/navigation/modals/cell-preview';
+import type {YTError} from '../../../../../@types/types';
+
+type ActionType<R = any> = ThunkAction<R, RootState, any, CellPreviewAction>;
+
+const PREVIEW_LIMIT = 16 * 1024 * 1024; // 16MiB;
+
+const openCellPreview = (): CellPreviewAction => {
+    return {
+        type: CELL_PREVIEW.PARTITION,
+        data: {visible: true},
+    };
+};
+
+export const closeCellPreview = (): CellPreviewAction => {
+    return {
+        type: CELL_PREVIEW.PARTITION,
+        data: {visible: false},
+    };
+};
+
+export const setDataCellPreview = (data: any): CellPreviewAction => {
+    return {
+        type: CELL_PREVIEW.SUCCESS,
+        data: {data},
+    };
+};
+
+export const setPathCellPreview = (cellPath: string): CellPreviewAction => {
+    return {
+        type: CELL_PREVIEW.PARTITION,
+        data: {cellPath},
+    };
+};
+
+export const setErrorCellPreview = (error: YTError): CellPreviewAction => {
+    return {
+        type: CELL_PREVIEW.FAILURE,
+        data: {error},
+    };
+};
+
+export const requestCellPreview = (): CellPreviewAction => {
+    return {
+        type: CELL_PREVIEW.REQUEST,
+        data: undefined,
+    };
+};
+
+export const showCellPreviewModal = (columnName: string, index: number): ActionType => {
+    return async (dispatch, getState) => {
+        batch(() => {
+            dispatch(requestCellPreview());
+            dispatch(openCellPreview());
+        });
+
+        const path = getPath(getState());
+
+        const cellPath = `${path}{${columnName}}[#${index}:#${index + 1}]`;
+
+        let cell: any;
+        let error: YTError;
+
+        try {
+            const json = await ytApiV4.readTable({
```

Comment:
The request might took some time and it is better to use cancellation.
Please see examples of usage of `CancelHelper`.
And probably we need to cancel such requests when user closes the dialog before finish of loading.